### PR TITLE
Fixed remaining clipping issues

### PR DESCRIPTION
**Pull request by carldebilly**
GitHub Issues:
* #2083

# Bugfix

## What is the current behavior?
Sometimes some controls were not clipped properly when they are supposed to be. It's happening when the control is trying to take more space than available size.

## What is the new behavior?
The logic has been improved to follow closer the Windows' behavior.

## PR Checklist

- [X] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [X] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: doc/ReleaseNotes/_ReleaseNotes.md [hunk 1]
```diff
@@ -192,6 +192,7 @@
 * #1708 Fix initial Flyout placement and window resize placement
 * [Android] #2007 ComboBox does not take Window.VisibleBounds to position its popup
 * [Wasm] Fixes the measure of a TextBoxView #2034 #2095
+* [Android] [Wasm] Recent clipping improvements were incompleted. Fixed a case where a control was allowed to draw itself to use more than available place in the _arrange_ phase.
```
Comment (jeromelaban) [inline doc/ReleaseNotes/_ReleaseNotes.md]: Not needed.

File: src/Uno.UI/UI/Xaml/FrameworkElement.Layout.wasm.cs [hunk 2]
```diff
@@ -149,13 +151,12 @@ private void InnerArrangeCore(Rect finalRect)
...
 			{
-				if (arrangeSize.Width < _unclippedDesiredSize.Width - SIZE_EPSILON)
+				if (IsLessThanAndNotCloseTo(arrangeSize.Width, _unclippedDesiredSize.Width))
 				{
 					_logDebug?.Debug($"{DepthIndentation}{this}: (arrangeSize.Width) {arrangeSize.Width} < {_unclippedDesiredSize.Width}: NEEDS CLIPPING.");
 					needsClipToSlot = true;
 				}
-
-				if (arrangeSize.Height < _unclippedDesiredSize.Height - SIZE_EPSILON)
+				else if (IsLessThanAndNotCloseTo(arrangeSize.Height, _unclippedDesiredSize.Height))
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/FrameworkElement.Layout.wasm.cs]: Not sure if it's expected, but you changed this the opposite way in the _Layouter.cs_ (removed the `else`)

File: doc/ReleaseNotes/_ReleaseNotes.md (hunk 1 above)
Comment (carldebilly) [reply doc/ReleaseNotes/_ReleaseNotes.md]: Should I remove it?

File: src/Uno.UI/UI/Xaml/FrameworkElement.Layout.wasm.cs [hunk 3]
```diff
@@ -149,13 +151,12 @@ private void InnerArrangeCore(Rect finalRect)
 
 			if (allowClipToSlot && !needsClipToSlot)
 			{
-				if (arrangeSize.Width < _unclippedDesiredSize.Width - SIZE_EPSILON)
+				if (IsLessThanAndNotCloseTo(arrangeSize.Width, _unclippedDesiredSize.Width))
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/FrameworkElement.Layout.wasm.cs]: The `arrangeSize.Width` is not updated here?

Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/FrameworkElement.Layout.wasm.cs]: Nope... there's 2 phases very similar, that's the first phase.

File: src/Uno.UI/UI/Xaml/FrameworkElement.Layout.wasm.cs (hunk 2 above)
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/FrameworkElement.Layout.wasm.cs]: You're comparing phase 1 with phase 2...  there's no disrutpancies :-)

**Pull request merged**